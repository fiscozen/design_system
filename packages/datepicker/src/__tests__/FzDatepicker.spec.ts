import { describe, it, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { FzDatepicker } from "..";

describe.concurrent("FzDatepicker", () => {
  vi.mock("@fiscozen/composables", () => ({
    useBreakpoints: vi.fn().mockReturnValue({
      isSmaller: vi.fn().mockReturnValue(false),
    }),
  }));
  it("matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches range snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        range: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches multicalendars range snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        range: true,
        multiCalendars: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches weekpicker snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        weekPicker: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches monthpicker snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        monthPicker: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches yearpicker snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        yearPicker: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches yearpicker snapshot", async ({ expect }) => {
    const today = new Date();
    const tomorrow = new Date(today);
    tomorrow.setDate(tomorrow.getDate() + 1);
    const afterTomorrow = new Date(tomorrow);
    afterTomorrow.setDate(tomorrow.getDate() + 1);

    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        disabledDates: [tomorrow, afterTomorrow],
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches inline timepicker snapshot", async ({ expect }) => {
    const wrapper = mount(FzDatepicker, {
      props: {
        modelValue: new Date(),
        timePickerInline: true,
        enableTimePicker: true,
        enableMinutes: true,
        is24: true,
      },
      slots: {},
    });

    await wrapper.find("input").trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });
});
