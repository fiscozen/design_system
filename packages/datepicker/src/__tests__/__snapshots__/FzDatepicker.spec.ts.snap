// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzDatepicker > matches snaphost 1`] = `
"<div class="dp__main dp__theme_light" data-datepicker-instance="">
  <div>
    <!---->
    <div class="dp__input_wrap">
      <!----><input data-test="dp-input" class="dp__pointer dp__input_readonly dp__input dp__input_icon_pad dp__input_invalid dp__input_reg" inputmode="none" placeholder="" autocomplete="off" aria-label="Datepicker input" aria-invalid="true">
      <div>
        <!----><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" fill="currentColor" aria-hidden="true" class="dp__icon dp__input_icon dp__input_icons">
          <path d="M29.333 8c0-2.208-1.792-4-4-4h-18.667c-2.208 0-4 1.792-4 4v18.667c0 2.208 1.792 4 4 4h18.667c2.208 0 4-1.792 4-4v-18.667zM26.667 8v18.667c0 0.736-0.597 1.333-1.333 1.333 0 0-18.667 0-18.667 0-0.736 0-1.333-0.597-1.333-1.333 0 0 0-18.667 0-18.667 0-0.736 0.597-1.333 1.333-1.333 0 0 18.667 0 18.667 0 0.736 0 1.333 0.597 1.333 1.333z"></path>
          <path d="M20 2.667v5.333c0 0.736 0.597 1.333 1.333 1.333s1.333-0.597 1.333-1.333v-5.333c0-0.736-0.597-1.333-1.333-1.333s-1.333 0.597-1.333 1.333z"></path>
          <path d="M9.333 2.667v5.333c0 0.736 0.597 1.333 1.333 1.333s1.333-0.597 1.333-1.333v-5.333c0-0.736-0.597-1.333-1.333-1.333s-1.333 0.597-1.333 1.333z"></path>
          <path d="M4 14.667h24c0.736 0 1.333-0.597 1.333-1.333s-0.597-1.333-1.333-1.333h-24c-0.736 0-1.333 0.597-1.333 1.333s0.597 1.333 1.333 1.333z"></path>
        </svg>
      </div>
      <!---->
      <!---->
    </div>
  </div>
  <div>
    <transition-stub name="" appear="false" persisted="false" css="false">
      <!---->
    </transition-stub>
  </div>
</div>"
`;
